src/xenialdan/PocketRadio/task/SongPlayerTask.php

```diff
@@ -18,31 +18,29 @@ class SongPlayerTask extends Task
 {
     public $song = null;
     public $songfilename = "";
-    /** @var Plugin|Loader */
+    /** @var PluginBase */
     public $owner;
     /** @var bool */
     protected $playing = false;
     /** @var int */
     private $tick = -1;
 
-    public function __construct(Plugin $owner, string $songfilename, Song $song)
+    public function __construct(PluginBase $owner, string $songfilename, Song $song)
     {
         $this->owner = $owner;
         $this->song = $song;
         $this->songfilename = $songfilename;
-        Loader::$tasks[] = $this->getTaskId();
+        Loader::$tasks[] = $this;
```

Comment:
what the heck???????????????????????????????????????????????????????????????????

Reply:
at least use handler id or something

Reply:
`Task:getId()` was removed from PM4 so this is how to store the Task.

Reply:
I did not find anything similar to an ID!

Reply:
see TaskHandler